drange/lib/DataStructure.js

```diff
@@ -0,0 +1,98 @@
+"use strict";
```

Comment:
> "we don't generate discontinuous-ranges"

That is a good point. It never occurred to me that the name is a misnomer because we are generating an integer _from_ a discontinuous range. It should like we should rename the method. What do you purpose?

> Is this original code?

Based on your comment on jsverify/jsverify!19 to "keep jsverify lib as dependency free as possible," I reimplemented [discontinuous-range](https://www.npmjs.com/package/discontinuous-range) from scratch to have the minimal set of features required for this generator. I found `jsverify/lib/finitemap.js` as an existing data structure in `jsverify` and followed that as a guide. I was surprised to see that it was not immutable. I have been trying to use immutable data-structures as much as possible (in my other work), but I though that following the existing conventions were more important, so I didn't make it immutable.

All of this being said I think these are the _reasonable_ choices based on your feedback. Which do you want me to pursue?

1) Use the *mutable* [discontinuous-range](https://www.npmjs.com/package/discontinuous-range) package as a dependency and scrap this work.
2) Rewrite DataStructure.js to be immutable, publish it as a separate npm package, and use it as a dependency.